sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/ErrorParserSuite.scala

```diff
@@ -255,4 +255,17 @@ class ErrorParserSuite extends AnalysisTest {
         |SELECT b
       """.stripMargin, 2, 9, 10, msg + " test-table")
   }
+
+  test("datatype not supported") {
+    // general bad types
+    intercept("SELECT cast(1 as badtype)", 1, 17, 17, "DataType badtype is not supported.")
+
+    // special handling on char and varchar
```

Comment:
Can we put the tests in `QueryParsingErrorsSuite`? Which has the infra to check error class

Reply:
This class also has the `intercept` that can check the error class (see the below `PARSE_CHAR_MISSING_LENGTH`). The current tests check whether the error class is `PARSE_CHAR_MISSING_LENGTH`.
Wonder what's the difference between this `ErrorParserSuite` and `QueryParsingErrorsSuite`? Seems they all testing parsing exceptions, but this `ErrorParserSuite` also checks the ANTLR errors.